hpx/util/external_timer.hpp

```diff
@@ -0,0 +1,318 @@
+//  Copyright (c) 2007-2016 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once    // prevent multiple inclusions of this header file.
+
+#include <hpx/config.hpp>
+#include <hpx/assertion.hpp>
+#include <hpx/coroutines/thread_id_type.hpp>
+#include <hpx/runtime/threads/thread_data_fwd.hpp>
+#include <hpx/util/thread_description.hpp>
+#include <cstdint>
+#include <memory>
+#include <string>
+
+/* note carefully...some symbols are referenced by annotated_function.hpp
+ * which gets included by many things.  Therefore, we include a weak reference
+ * so that we don't get linker errors for shared object libraries that don't
+ * link with libhpx.so. */
+
+#if defined(_WIN32) || defined(__WIN32__) || defined(WIN32)
+#define HPX_UTIL_WEAK_PRE __declspec(selectany)
+#define HPX_UTIL_WEAK_POST
+#else // not WIN32
+#ifdef __clang__
+#define HPX_UTIL_WEAK_PRE
+#define HPX_UTIL_WEAK_POST __attribute__((weak_import))
+#else // not clang
+#define HPX_UTIL_WEAK_PRE __attribute__((weak))
+#define HPX_UTIL_WEAK_POST
+#endif
+#endif
```

Comment:
Where are these needed/used? I don't see any signs of them here or in APEX.

Reply:
those are leftover from an aborted solution that wasn't needed (solved with HPX_EXPORT). I will remove them.